test/integration/oauth_server_headers_test.go

```diff
@@ -68,18 +70,35 @@ func checkNewReqHeaders(t *testing.T, rt http.RoundTripper, check_url string) {
 		t.Fatalf("unexpected error %v", err)
 	}
 
-	checkImportantHeaders := map[string]string{
+	importantHeaders := http.Header{}
```

Comment:
these are not important heasders only, it's all headers now ... I would change the name accordingly.